### ISSUE TITLE
Add testing for OCI hooks

### ISSUE THREAD

**Issue by sondavidb** [containerd/nerdctl#3156]
### What is the problem you're trying to solve

Looking in pkg/ocihook, we don't seem to have to have any tests to ensure these OCI hooks are working as intended. We have one test in [container_run_test.go](https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/container_run_test.go#L487), but nothing more extensive than that.

### Describe the solution you'd like

Add more extensive testing to ensure OCI hooks are working properly.

**Comment by apostasie** [comment]
Yes, we need this one way or the other (@AkihiroSuda can we tag this "CI area" or "testing" or something in the line?).

I have been doing work recently inside oci-hooks, and debugging anything in there, at the intersection of containerd/runc/cni-plugins/oci-hooks is a major brainf.
